test-interface/src/main/scala/org/scalajs/testinterface/internal/TaskInfoBuilder.scala

```diff
@@ -0,0 +1,24 @@
+package org.scalajs.testinterface.internal
+
+import sbt.testing._
+
+import org.scalajs.testcommon.{TaskInfo, Serializer}
+
+private[internal] object TaskInfoBuilder {
+  def detachTask(task: Task, runner: Runner): TaskInfo = {
```

Comment:
Because they attach/associate with the runner. The task might need to report things to a runner, so it needs to retain a reference to it. I recommend you read the Scaladoc of `serializeTask` and `deserializeTask`.